### PR TITLE
Add InternVL3.5 model support

### PR DESCRIPTION
**Pull request by houleisai**
```markdown
### PR 类型

新模型支持 / Models

### PR 变更

新增 InternVL3.5-1B 在 PaddleFormers 中的模型接入，包括配置、模型、Processor、ImageProcessor、Auto 注册、单测与验证脚本。

### 主要内容

1. 新增 `paddleformers/transformers/internvl3_5`
   - `InternVLChatConfig`
   - `InternVisionConfig`
   - `InternVLChatModel`
   - `InternVisionModel`
   - `InternVLProcessor`
   - `InternVLImageProcessor`

2. 接入 Auto 入口
   - `AutoConfig`
   - `AutoModel`
   - `AutoProcessor`
   - `AutoImageProcessor`

3. 支持从原始 HuggingFace 权重加载
   - 原始模型路径：`nternVL3_5-1B`
   - 支持 `convert_from_hf=True`
   - 已验证 `model.safetensors` 权重可完整加载

4. 新增测试
   - `tests/transformers/internvl3_5/test_modeling.py`
   - `tests/transformers/internvl3_5/test_processor.py`

### 验证结果

#### 1. 单卡前向精度对齐

使用 FP32 对齐 transformers 与 PaddleFormers 前向 logits：

```text
logits shape: (1, 264, 151936)
logits mean abs diff: 0.00002021
logits p95 abs diff:  0.00006688
logits p99 abs diff:  0.00014514
logits max abs diff:  0.00109911
```
满足 `1e-2` 量级要求。

#### 2. 生成结果对齐

使用 greedy search，`max_new_tokens=10`，与 transformers 生成前 10 个 token 完全一致：

```text
torch tokens:
[1096, 374, 264, 36190, 315, 264, 2766, 1809, 13, 576]

paddle tokens:
[1096, 374, 264, 36190, 315, 264, 2766, 1809, 13, 576]

first10_equal: True
```

#### 3. GSM8K 300 步训练 loss

loss 整体呈下降趋势，训练过程正常，无 NaN/Inf。

#### 4. 单测结果

```bash
../miniconda3/envs/paddleformers/bin/python -m unittest \
  tests.transformers.internvl3_5.test_modeling \
  tests.transformers.internvl3_5.test_processor
```

结果：

```text
Ran 12 tests in 1.220s
OK
```

### CODE REVIEW

File: paddleformers/transformers/internvl3_5/processor.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            tokenizer.convert_tokens_to_ids(img_context_token) if tokenizer is not None else 151671
+        )
+        super().__init__(image_processor, tokenizer, chat_template=chat_template)
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
+        from ..auto.tokenizer import AutoTokenizer
+        from .image_processor import InternVLImageProcessor
+
+        tokenizer = AutoTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/processor.py]: P1 这里覆盖了基类加载流程，却只分别加载 tokenizer 和 image processor，随后用默认参数构造 processor。`OpenGVLab/InternVL3_5-1B` 的 `chat_template.jinja` 因此不会进入 `self.chat_template`，`AutoProcessor.from_pretrained(...)` 后调用 `apply_chat_template(...)` 会直接报“processor does not have a chat template”；`processor_config.json` 中的字段也会被忽略。请复用 `ProcessorMixin.from_pretrained` 的配置/模板加载流程，并补充从 checkpoint 经 `AutoProcessor` 加载后实际调用 `apply_chat_template` 的回归测试。

File: paddleformers/transformers/auto/image_processing.py [hunk 2]
```diff
@@ -62,6 +62,8 @@
         "qwen3_vl": ("Qwen3VLImageProcessor", "Qwen3VLImageProcessorFast"),
         "glm_ocr": ("Glm46VImageProcessor"),
         "paligemma": ("PaliGemmaImageProcessor"),
+        "internvl_chat": ("InternVLImageProcessor"),
```
Comment (risemeup1111) [inline paddleformers/transformers/auto/image_processing.py:65]: P1 这条映射无法覆盖本 PR 要兼容的原始 checkpoint：`OpenGVLab/InternVL3_5-1B` 的 `preprocessor_config.json` 已显式声明 `image_processor_type=GotOcr2ImageProcessorFast`，而 `AutoImageProcessor.from_pretrained` 会优先按该字段选类，只有类型缺失时才回退到 model config mapping。因此这个 Auto 入口不会实例化 `InternVLImageProcessor`，也就不会产生动态切图所需的 `num_patches_list`。请为该 checkpoint 做明确的兼容分派，并增加 `AutoImageProcessor.from_pretrained` 类型及多切片输出回归。

File: paddleformers/transformers/internvl3_5/image_processor.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        pixel_values = []
+        num_patches_list = []
+        for image in images:
+            pil_image = _to_pil_image(image)
+            tiles = self.dynamic_preprocess(
+                pil_image,
+                min_num=kwargs.pop("min_patches", self.min_patches),
+                max_num=kwargs.pop("max_patches", self.max_patches),
+                image_size=kwargs.pop("image_size", self.size["height"]),
+                use_thumbnail=kwargs.pop("use_thumbnail", self.use_thumbnail),
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/image_processor.py]: P1 这些调用参数在逐图循环内从同一个 `kwargs` 中 `pop`：第一张图之后键已被删除，后续图片会退回实例默认值。例如实例默认 `max_patches=12,use_thumbnail=True` 时，调用 `processor([img1, img2], max_patches=1, use_thumbnail=False)` 只会约束第一张图，第二张仍可能产生 13 个 tile，导致同一 batch 的预处理语义和视觉 token 数错误。请在循环外解析一次参数并对每张图复用，同时增加多图且调用参数不同于默认值的回归。

File: paddleformers/transformers/internvl3_5/modeling.py [hunk 4]
```diff
@@ -0,0 +1,763 @@
...
+            if output_hidden_states:
+                all_hidden_states += (hidden_states,)
+            hidden_states = layer(hidden_states, attention_mask=causal_mask, position_ids=position_ids)
+        hidden_states = self.norm(hidden_states)
+        if output_hidden_states:
+            all_hidden_states += (hidden_states,)
+        if not return_dict:
+            return tuple(v for v in [hidden_states, None, all_hidden_states] if v is not None)
+        return BaseModelOutputWithPast(
+            last_hidden_state=hidden_states, past_key_values=None, hidden_states=all_hidden_states
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/modeling.py]: P1 当前语言模型无论 `use_cache` 取值都固定返回 `past_key_values=None`，attention/decoder 也没有接收或更新传入的 cache。生成框架只有拿到非空 cache 才会在下一步把 `input_ids` 截为最后一个 token；这里会让每个解码步重新计算包含 256×tile 个视觉 token 的完整前缀。原始 InternVL 生成路径明确以 `use_cache=True` 调用语言模型，请实现并透传 Qwen3 KV cache，并用多步生成测试断言首步产生 cache、后续步只处理新增 token。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 状态 |
| --- | --- | --- | --- |
| 1 | [`processor.py:55`](https://github.com/PaddlePaddle/PaddleFormers/pull/4613#discussion_r3754781256) | P1 | ✅ |
| 2 | [`auto/image_processing.py:65`](https://github.com/PaddlePaddle/PaddleFormers/pull/4613#discussion_r3754782222) | P1 | ✅ |
| 3 | [`image_processor.py:151`](https://github.com/PaddlePaddle/PaddleFormers/pull/4613#discussion_r3754783250) | P1 | ✅ |
| 4 | [`modeling.py:215`](https://github.com/PaddlePaddle/PaddleFormers/pull/4613#discussion_r3754784224) | P1 | ✅ |
| 5 | [`qwen3/modeling.py:34`](https://github.com/PaddlePaddle/PaddleFormers/pull/4613#discussion_r3754916920) | P1 | ✅ |
| 6 | [`internvl3_5/modeling.py:478`](https://github.com/PaddlePaddle/PaddleFormers/pull/4613#discussion_r3754917973) | P1 | ✅ |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

File: paddleformers/transformers/qwen3/modeling.py [hunk 5]
```diff
@@ -31,7 +31,19 @@
 from paddle.distributed.fleet.recompute.recompute import recompute
 from paddle.distributed.fleet.utils.sequence_parallel_utils import ScatterOp
 
-from paddleformers.transformers.gpt_provider import GPTModelProvider
+try:
+    from paddleformers.transformers.gpt_provider import GPTModelProvider
+except ModuleNotFoundError as exc:
```
Comment (risemeup1111) [inline paddleformers/transformers/qwen3/modeling.py]: P1 这里的 fallback 在未安装 PaddleFleet 时不会生效：`gpt_provider.py:31-36` 在 `is_paddlefleet_available()` 为 false 时显式抛出普通 `ImportError`，而 PaddleFleet 仅由 `setup.py` 的可选 extra 安装；该异常不会被 `ModuleNotFoundError` 捕获，默认安装下 `InternVLChatModel` 会在导入 `Qwen3ForCausalLMDeprecated` 时直接失败。请按可用性分支选择 provider/fallback（或捕获并限定这里实际的 `ImportError`），并补充无 PaddleFleet 环境下导入 deprecated 实现的回归。

File: paddleformers/transformers/internvl3_5/modeling.py [hunk 6]
```diff
@@ -0,0 +1,589 @@
...
+            input_embeds = self._merge_visual_embeds(input_ids, input_embeds, vit_embeds)
+
+        outputs = self.language_model(
+            input_ids=None,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            past_key_values=past_key_values,
+            inputs_embeds=input_embeds,
+            labels=labels,
+            use_cache=use_cache,
```
Comment (risemeup1111) [inline paddleformers/transformers/internvl3_5/modeling.py:478]: P1 `output_hidden_states` 仍是 `InternVLChatModel.forward` 的公开参数，但这里不再把它传给语言模型；当前共享 `Qwen3Model.forward` 也没有收集各层状态，因此 `model(..., output_hidden_states=True)` 会静默返回 `outputs.hidden_states is None`。上一版对应回归测试也在本提交中被删除，依赖层特征的调用会被破坏。请为共享 Qwen3 路径实现并透传 hidden-state 收集，同时恢复该测试。

File: paddleformers/transformers/internvl3_5/processor.py (hunk 1 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/processor.py]: 实现侧已改为复用 `ProcessorMixin.from_pretrained`，模板加载路径已恢复；但新增的 `test_auto_processor_real_checkpoint_chat_template` 依赖硬编码本机目录并通过 `skipUnless` 跳过，常规 CI 仍不会执行原先要求的 `AutoProcessor -> apply_chat_template` 回归。请改为可在 CI 中运行的临时 checkpoint/fixture 后再关闭此项。

File: paddleformers/transformers/auto/image_processing.py (hunk 2 above)
Comment (risemeup1111) [reply paddleformers/transformers/auto/image_processing.py:65]: 已修复：当前 head 会在 `GotOcr2ImageProcessor[Fast] + InternVLProcessor` 组合下明确分派到 `InternVLImageProcessor`，并新增自包含的 AutoImageProcessor 多切片回归。

File: paddleformers/transformers/internvl3_5/image_processor.py (hunk 3 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/image_processor.py]: 已修复：调用级参数已在图像循环外解析并复用于每张图，新增多图且覆盖默认值的回归验证了 `[1, 1]` 切片结果。

File: paddleformers/transformers/internvl3_5/modeling.py (hunk 4 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/modeling.py]: 已修复：语言模型已切换到共享 Qwen3 cache 实现，首步返回 cache，生成输入准备会在后续步仅保留新增 token，并清空视觉特征。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

File: paddleformers/transformers/qwen3/modeling.py (hunk 5 above)
Comment (risemeup1111) [reply paddleformers/transformers/qwen3/modeling.py]: `90b16984` 删除了上一版 guard，当前又在 `qwen3/modeling.py:34` 无条件导入 `gpt_provider`；而 `gpt_provider.py:31-36` 在默认未安装可选 PaddleFleet 时仍抛出 `ImportError`。因此该项仍未修复：请将 provider 导入与 deprecated 单卡实现解耦，并增加无 PaddleFleet 环境下的导入回归。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

File: paddleformers/transformers/qwen3/modeling.py (hunk 5 above)
Comment (risemeup1111) [reply paddleformers/transformers/qwen3/modeling.py]: 已修复：`qwen3/modeling.py:34-45` 现在同时处理 `gpt_provider.py` 显式抛出的 `ImportError`，deprecated 单卡实现可在缺少 PaddleFleet 时正常导入；`tests/transformers/qwen3/test_modeling.py:411-442` 新增了隔离子进程回归，并验证 provider 实际使用时仍会给出依赖错误。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

File: paddleformers/transformers/internvl3_5/processor.py (hunk 1 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/processor.py]: 已修复：`tests/transformers/internvl3_5/test_processor.py:116-144` 现在用临时 tokenizer、processor/image processor 配置和 `chat_template.jinja` 构造自包含 checkpoint，常规测试会完整执行 `AutoProcessor.from_pretrained -> apply_chat_template`，不再依赖本机目录或跳过条件。

File: paddleformers/transformers/internvl3_5/modeling.py (hunk 6 above)
Comment (risemeup1111) [reply paddleformers/transformers/internvl3_5/modeling.py:478]: 已修复：`internvl3_5/modeling.py:471-480` 已透传 `output_hidden_states`，共享 Qwen3 路径在 `qwen3/modeling.py:598-712` 收集每层输入及最终归一化状态，并由 `tests/transformers/internvl3_5/test_modeling.py:140-151` 验证数量和形状。

**Review by risemeup1111 — APPROVED** [review]
